src/internal/ScrollCaptor.js

```diff
@@ -18,41 +18,67 @@ class ScrollCaptor extends Component<CaptorProps> {
   scrollTarget: HTMLElement;
   touchStart: number;
 
+  // Scroll element previous state variables
+  previousScrollTop: number = 0;
+  previousScrollHeight: number = 0;
+  previousClientHeight: number = 0;
+
+  // Event listener flags
+  listeningToScroll: boolean = false;
+  listeningToTouchStart: boolean = false;
+  listeningToTouchMove: boolean = false;
+
   componentDidMount() {
-    this.startListening(this.scrollTarget);
+    this.previousScrollHeight = this.scrollTarget.scrollHeight;
+    this.previousClientHeight = this.scrollTarget.previousClientHeight;
+    this.refreshListening(this.scrollTarget);
+  }
+
+  componentDidUpdate() {
+    if (this.previousScrollHeight !== this.scrollTarget.scrollHeight || this.previousClientHeight !== this.scrollTarget.clientHeight) {
+      this.previousScrollHeight = this.scrollTarget.scrollHeight;
+      this.previousClientHeight = this.scrollTarget.clientHeight;
+      this.refreshListening(this.scrollTarget);
+    }
   }
+
   componentWillUnmount() {
     this.stopListening(this.scrollTarget);
   }
-  startListening(el: HTMLElement) {
+  refreshListening(el: HTMLElement) {
     // bail early if no scroll available
-    if (!el) return;
-    if (el.scrollHeight <= el.clientHeight) return;
+    if (!el || el.scrollHeight <= el.clientHeight)  {
+      this.stopListening(el);
+      return;
+    }
 
     // all the if statements are to appease Flow 😢
-    if (typeof el.addEventListener === 'function') {
-      el.addEventListener('wheel', this.onWheel, false);
+    if (typeof el.addEventListener === 'function' &&  !this.listeningToScroll) {
+      el.addEventListener('scroll', this.onScroll, false);
+      this.listeningToScroll = true;
     }
-    if (typeof el.addEventListener === 'function') {
+    if (typeof el.addEventListener === 'function' && !this.listeningToScroll) {
```

Comment:
this should be listeningToTouchStart, right?

Reply:
Yep, fixed.